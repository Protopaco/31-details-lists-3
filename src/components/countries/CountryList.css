.countryElements{
    display: flex;
    flex-wrap: wrap;
    list-style: none;
}

a{
    text-decoration: none;
}

.controls li{
    display: inline;
    margin: 10px;
    color: brown;
}

button{
    background-color: brown;
    color: beige;
    width: 100px;
    height: 25px;
    border: none;
    outline: none;
    filter: drop-shadow(1px 1px 1px tan)
}

button:active{
    filter: drop-shadow(.5px .5px .5px tan)
}

div ul{
    display: flex;
    justify-content: center;
}