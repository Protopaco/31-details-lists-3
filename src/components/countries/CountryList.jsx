import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Country from './Country'

const CountryList = ({ countries }) => {
    const countryElements = countries.map(country => (
        <li key={country.code}>
            <Link to={`/country/${counrty.code}`}>
                <Country flagURL={country.flagURL} name={country.name} />
            </Link>
        </li>
    ));

    return (
        <ul data-testid="countries">
            {countryElements}
        </ul>
    );
};

CountryList.propTypes = {
    counrties: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        flagURL: PropTypes.string.isRequired
    })).isRequired
};
export default CountryList;
