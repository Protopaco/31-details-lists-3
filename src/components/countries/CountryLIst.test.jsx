import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import CountryList from './CountryList';

describe('CountryList component', () => {
    afterEach(() => cleanup());
    it('renders CountryList', () => {
        const { asFragment } = render(
            <Router>
                <CountryList
                    countries={[{ "name": 'f', "code": 'f', "flagURL": 'f' }]}
                    changePage={() => console.log('f')}
                    page={'f'} />
            </Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
