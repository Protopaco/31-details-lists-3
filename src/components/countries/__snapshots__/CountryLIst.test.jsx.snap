// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList component renders CountryList 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="controls"
    >
      <li>
        <button>
          &lt;
        </button>
      </li>
      <li
        data-testid="pageNum"
      >
        f
      </li>
      <li>
        <button
          data-testid="advancePage"
        >
          &gt;
        </button>
      </li>
    </ul>
    <ul
      class="countryElements"
      data-testid="countries"
    >
      <li>
        <a
          href="/country/f"
        >
          <figure
            class="figure"
          >
            <img
              alt="f"
              class="image"
              src="f"
            />
            <figcaption
              class="caption"
            >
              f
            </figcaption>
          </figure>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
